x/concentrated-liquidity/incentives_test.go

```diff
@@ -401,6 +435,84 @@ func (s *KeeperTestSuite) TestCalcAccruedIncentivesForAccum() {
 			},
 			expectedPass: true,
 		},
+		"two incentive records with same denom, different start time": {
+			poolId:              defaultPoolId,
+			accumUptime:         types.SupportedUptimes[0],
+			qualifyingLiquidity: sdk.NewDec(100),
+			timeElapsed:         time.Hour,
+
+			poolIncentiveRecords: []types.IncentiveRecord{incentiveRecordOne, incentiveRecordOneWithDifferentStartTime},
+
+			expectedResult: sdk.DecCoins{
+				// We expect both incentive record to qualify
+				expectedIncentives(incentiveRecordOne.IncentiveDenom, incentiveRecordOne.EmissionRate.Add(incentiveRecordOneWithDifferentStartTime.EmissionRate), time.Hour, sdk.NewDec(100)), // since we have 2 records with same denom, the rate of emission went up x2
+			},
+			expectedIncentiveRecords: []types.IncentiveRecord{
+				// We only going to charge both incentive records
+				chargeIncentive(incentiveRecordOne, time.Hour),
+				chargeIncentive(incentiveRecordOneWithDifferentStartTime, time.Hour),
+			},
+			expectedPass: true,
+		},
+		"two incentive records with different denom, different start time and same uptime": {
+			poolId:              defaultPoolId,
+			accumUptime:         types.SupportedUptimes[0],
+			qualifyingLiquidity: sdk.NewDec(100),
+			timeElapsed:         time.Hour,
+
+			poolIncentiveRecords: []types.IncentiveRecord{incentiveRecordOneWithDifferentStartTime, incentiveRecordOneWithDifferentDenom},
+
+			expectedResult: sdk.DecCoins{
+				// We expect both incentive record to qualify
+				expectedIncentives(incentiveRecordOneWithDifferentStartTime.IncentiveDenom, incentiveRecordOne.EmissionRate, time.Hour, sdk.NewDec(100)), // since we have 2 records with same denom, the rate of emission went up x2
+				expectedIncentives(incentiveRecordOneWithDifferentDenom.IncentiveDenom, incentiveRecordOne.EmissionRate, time.Hour, sdk.NewDec(100)),     // since we have 2 records with same denom, the rate of emission went up x2
+			},
+			expectedIncentiveRecords: []types.IncentiveRecord{
+				// We only charge the first incentive record because the second minUpTime hasn't been hit yet
+				chargeIncentive(incentiveRecordOneWithDifferentStartTime, time.Hour),
+				chargeIncentive(incentiveRecordOneWithDifferentDenom, time.Hour),
+			},
+			expectedPass: true,
+		},
+		"two incentive records with same denom, different min up-time": {
+			poolId:              defaultPoolId,
+			accumUptime:         types.SupportedUptimes[0],
+			qualifyingLiquidity: sdk.NewDec(100),
+			timeElapsed:         time.Hour,
+
+			poolIncentiveRecords: []types.IncentiveRecord{incentiveRecordOne, incentiveRecordOneWithDifferentMinUpTime},
+
+			expectedResult: sdk.DecCoins{
+				// We expect both incentive record to qualify
+				expectedIncentives(incentiveRecordOne.IncentiveDenom, incentiveRecordOne.EmissionRate, time.Hour, sdk.NewDec(100)), // since we have 2 records with same denom, the rate of emission went up x2
```

Comment:
is this correct? only one seems to qualify
```suggestion
				expectedIncentives(incentiveRecordOne.IncentiveDenom, incentiveRecordOne.EmissionRate, time.Hour, sdk.NewDec(100)),

Reply:
yea i forgot to delete the comment. the rate of emission shouldn't be x2